fhir-server/src/main/java/org/linuxforhealth/fhir/server/listener/FHIRServletContextListener.java

```diff
@@ -299,6 +304,17 @@ public void contextInitialized(ServletContextEvent event) {
 
             // Now init is complete, tell all registered callbacks
             EventManager.serverReady(serviceManagerId);
+            
+            // Set the configured tenants list which will be used in FHIRRestServletFilter to check 
+            //   if the tenant Id is present in the list of configured tenants and return the tenant Id from the configured tenant list. 
+            // This is done to avoid Uncontrolled user input data used in path expression
+            List<String> configuredTenantsList = fhirConfiguration.getConfiguredTenants();
+            if (configuredTenantsList.isEmpty()) {
+                throw new IllegalStateException("No configured tenants were found");
+            }
+            configuredTenants = Collections.unmodifiableMap(configuredTenantsList.stream().collect(Collectors.toConcurrentMap(tenantId -> tenantId, tenantId -> tenantId)));
```

Comment:
I don't think it needs to be a concurrent map, but maybe there's something I'm missing?

Reply:
The map is being initialized in one thread but used in other threads. Without explicit synchronization, it's safer to use a ConcurrentHashMap here (although the act of making it unmodifiable may offer the same protection....would be interesting to look at this).